src/firefly/js/visualize/VisUtil.js

```diff
@@ -734,7 +750,131 @@ export function formatFluxValue(value) {
+//
+//     return highPts[0];
+// }
+
+
+export function getTopmostVisiblePoint(pv,xOff, yOff) {
+    const plot= primePlot(pv);
+    const cc= CysConverter.make(plot);
+    const ipt= cc.getImageCoords(makeDevicePt(xOff,yOff));
+    // if (cc.pointInPlot(ipt)) return ipt;
+    // if (isImageCoveringArea(pv,ipt,xOff,yOff)) return ipt;
+    if (isImageCoveringArea(pv,ipt,2,2)) return ipt;
+
+
+    const {dataWidth,dataHeight}= plot;
+    const {viewDim} = pv;
+
+    const lineSegs= [
+        {pt1: cc.getDeviceCoords(makeImagePt(0,0)), pt2: cc.getDeviceCoords(makeImagePt(dataWidth,0))},
+        {pt1: cc.getDeviceCoords(makeImagePt(dataWidth,0)), pt2: cc.getDeviceCoords(makeImagePt(dataWidth,dataHeight))},
+        {pt1: cc.getDeviceCoords(makeImagePt(dataWidth,dataHeight)), pt2: cc.getDeviceCoords(makeImagePt(0,dataHeight))},
+        {pt1: cc.getDeviceCoords(makeImagePt(0,dataHeight)), pt2: cc.getDeviceCoords(makeImagePt(0,0))}
+    ];
+
+    const foundSegs= lineSegs
+        .filter((lineSeg) => {
+                 const {pt1,pt2}= lineSeg;
+                 const iPt= findIntersectionPt(pt1.x,pt1.y,pt2.x,pt2.y, 0,0,viewDim.width-1,0);
+                 return iPt && iPt.onSeg1 && iPt.onSeg2;
+             })
+        .sort( (l1, l2) => l1.pt1.x - l2.pt1.x);
+
+    if (foundSegs[0]) {
+        const pt= findIntersectionPt(foundSegs[0].pt1.x,foundSegs[0].pt1.y,foundSegs[0].pt2.x,foundSegs[0].pt2.y, 0,0,viewDim.width-1,0);
+        return makeDevicePt(pt.x+xOff, pt.y+yOff);
+    }
+
+    const zXoff= xOff/plot.zoomFactor;
+    const zYoff= xOff/plot.zoomFactor;
+
+    const tryPts= [
+        makeImagePt(1+zXoff,1+zXoff),
+        makeImagePt(plot.dataWidth-zXoff,1+zYoff),
+        makeImagePt(plot.dataWidth-zXoff,plot.dataHeight-zYoff),
+        makeImagePt(1+zXoff, plot.dataHeight-zYoff),
+    ];
+
+
+    const highPts= tryPts
+        .map( (p) => cc.getDeviceCoords(p) )
+        .filter( (p) => cc.pointOnDisplay(p))
+        .sort( (p1,p2) => p1.y!==p2.y ? p1.y - p2.y : p1.x - p2.x);
+
+    return highPts[0];
+}
+
+
+
 
+export function isImageCoveringArea(pv,pt, width,height) {
+
+    const {viewDim} = pv;
+    const plot= primePlot(pv);
+    const cc= CysConverter.make(plot);
+    pt= cc.getDeviceCoords(pt);
+    const testPts= [
+        makeDevicePt(pt.x,pt.y),
+        makeDevicePt(pt.x+width,pt.y),
+        makeDevicePt(pt.x+width,pt.y+height),
+        makeDevicePt(pt.x,pt.y+height),
+    ];
+
+    const polyPts= [
+        cc.getDeviceCoords(makeImagePt(1,1)),
+        cc.getDeviceCoords(makeImagePt(plot.dataWidth,1)),
+        cc.getDeviceCoords(makeImagePt(plot.dataWidth,plot.dataHeight)),
+        cc.getDeviceCoords(makeImagePt(1, plot.dataHeight))
+    ];
+
+
+    const polyPtsAsArray= polyPts.map( (p) => [p.x,p.y]);
+
+    return testPts.every( (p) => pointInPolygon([p.x,p.y], polyPtsAsArray));
+}
+
+export function findIntersectionPt(x1, y1, x2, y2, x3, y3, x4, y4) {
```

Comment:
We need jsdoc or comments here to tell which pairs of (x,y) make first and second segments